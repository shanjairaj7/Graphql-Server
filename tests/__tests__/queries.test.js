const { gql } = require("apollo-server");
const { createTestServer } = require("../testServer");

const userID = "60d6c18d2dc62607ac8a0648";

// user id 60d6c18d2dc62607ac8a0648
describe("Queries", () => {
  test("me", async () => {
    const ME = gql`
      {
        me {
          id
          email
        }
      }
    `;

    const { query } = createTestServer({
      user: {
        id: userID,
      },
      models: {
        User: {
          findOne() {
            return {
              id: userID,
              name: "Shanjai",
              email: "shanjai@gmail.com",
              password: "hashedPassword",
              invited: false,
              role: "MEMBER",
            };
          },
        },
      },
    });

    const result = await query({ query: ME });
    expect(result).toMatchSnapshot();
  });
  test("feed", async () => {
    const FEED = gql`
      {
        feed {
          id
          message
          createdAt
          updatedAt
          creator {
            id
            name
            email
            password
            invited
            role
          }
        }
      }
    `;

    const { query } = createTestServer({
      user: {
        id: userID,
      },
      models: {
        Post: {
          find() {
            return [
              {
                id: "1",
                message: "Test",
                createdAt: "123123947",
                updatedAt: "293647623",
                creator: {
                  id: userID,
                  name: "Shanjai",
                  email: "shanjai@gmail.com",
                  password: null,
                  invited: false,
                  role: "MEMBER",
                },
              },
              {
                id: "2",
                message: "New Message",
                createdAt: "472983748",
                updatedAt: "182637226",
                creator: {
                  id: userID,
                  name: "Shanjai",
                  email: "shanjai@gmail.com",
                  password: null,
                  invited: false,
                  role: "MEMBER",
                },
              },
            ];
          },
        },
      },
    });

    const result = await query({ query: FEED });
    expect(result).toMatchSnapshot();
  });
});
