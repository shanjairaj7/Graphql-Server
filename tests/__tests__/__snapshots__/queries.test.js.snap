// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Queries feed 1`] = `
Object {
  "data": null,
  "errors": Array [
    [GraphQLError: Cast to ObjectId failed for value "{
  id: '60d6c18d2dc62607ac8a0648',
  name: 'Shanjai',
  email: 'shanjai@gmail.com',
  password: null,
  invited: false,
  role: 'MEMBER'
}" (type Object) at path "_id" for model "User"],
  ],
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;

exports[`Queries me 1`] = `
Object {
  "data": Object {
    "me": Object {
      "email": "shanjai@gmail.com",
      "id": "60d6c18d2dc62607ac8a0648",
    },
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;
